firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/report/ReportUploader.java

```diff
@@ -96,14 +99,21 @@ public boolean uploadReport(Report report, boolean dataCollectionToken) {
       final CreateReportRequest requestData =
           new CreateReportRequest(organizationId, googleAppId, report);
 
-      final boolean sent = createReportCall.invoke(requestData, dataCollectionToken);
+      boolean sent;
+      // For now, send native reports to reports endpoint regardless of the setting.
+      if (isUsingReportsEndpoint || report.getType() == Report.Type.NATIVE) {
+        sent = createReportCall.invoke(requestData, dataCollectionToken);
 
-      Logger.getLogger()
-          .i(
-              Logger.TAG,
-              "Crashlytics report upload "
-                  + (sent ? "complete: " : "FAILED: ")
-                  + report.getIdentifier());
+        Logger.getLogger()
+            .i(
+                Logger.TAG,
+                "Crashlytics report upload "
+                    + (sent ? "complete: " : "FAILED: ")
+                    + report.getIdentifier());
+      } else {
+        Logger.getLogger().d(Logger.TAG, "Send to reports endpoint disabled. Removing report.");
+        sent = true;
```

Comment:
If it didn't actually send, should we change the name of this bool to `shouldDeleteReport` or something similar?

Reply:
We can, sure.